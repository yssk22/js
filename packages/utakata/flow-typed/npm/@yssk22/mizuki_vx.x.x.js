// flow-typed signature: f9c32c028e3d261d286301826fb04c27
// flow-typed version: <<STUB>>/@yssk22/mizuki_v^1.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@yssk22/mizuki'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@yssk22/mizuki' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@yssk22/mizuki/flow-typed/npm/flow-bin_v0.x.x' {
  declare module.exports: any;
}

declare module '@yssk22/mizuki/src/index' {
  declare module.exports: any;
}

// Filename aliases
declare module '@yssk22/mizuki/flow-typed/npm/flow-bin_v0.x.x.js' {
  declare module.exports: $Exports<'@yssk22/mizuki/flow-typed/npm/flow-bin_v0.x.x'>;
}
declare module '@yssk22/mizuki/index' {
  declare module.exports: $Exports<'@yssk22/mizuki'>;
}
declare module '@yssk22/mizuki/index.js' {
  declare module.exports: $Exports<'@yssk22/mizuki'>;
}
declare module '@yssk22/mizuki/src/index.js' {
  declare module.exports: $Exports<'@yssk22/mizuki/src/index'>;
}
