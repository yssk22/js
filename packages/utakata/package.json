{
  "name": "@yssk22/utakata",
  "version": "1.0.0",
  "description": "Redux Library",
  "author": "yssk22 <yssk22@gmail.com",
  "license": "MIT",
  "scripts": {},
  "devDependencies": {
    "@yssk22/mizuki": "^1.0.0",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.6.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "flow-bin": "^0.81.0",
    "flow-cli": "^0.0.0-pre",
    "jest": "^23.6.0",
    "jest-cli": "^23.6.0"
  }
}
