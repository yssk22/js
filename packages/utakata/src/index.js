/* @flow */
import { add } from '@yssk22/mizuki';

module.exports = {
  add
};
