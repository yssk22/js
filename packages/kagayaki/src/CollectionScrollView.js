/* @flow */
import React from 'react';
import { View, Text, Button } from 'react-native';
import ScrollView from './ScrollView';
import { rest } from '@yssk22/utakata';

const ErrorContainerStyle = {
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center'
};

type Props<T> = {
  settings: rest.ResourceSettings<T>,
  errorText: ?string,
  retryButtonText: ?string,
  renderResource: (v: T) => Node
};

class RestRefreshableScrollViewList<T> extends React.Component<rest.Props & Props<T>> {
  constructor(props: rest.Props & Props<T>) {
    super(props);
  }

  componentWillMount() {
    rest.Action.getCollection(this.props.settings).run();
  }

  render() {
    const { settings, errorText, retryButtonText, renderResource, ...otherProps } = this.props;
    const helper = new rest.Helper(settings, this.props.rest);
    const data = helper.getData();
    const refreshing = helper.isCollectionLoading();
    const onRefresh = () => {
      rest.Action.getCollection(this.props.settings).run();
    };
    const error = helper.getError();
    if (error !== null) {
      return (
        <View style={ErrorContainerStyle}>
          <Text>{errorText || error}</Text>
          <Button
            title={retryButtonText || 'リトライ'}
            onPress={() => {
              rest.Action.getCollection(this.props.settings).run();
            }}
          />
        </View>
      );
    }
    return (
      <ScrollView
        data={data}
        refreshing={refreshing}
        onRefresh={onRefresh}
        childComponentFunc={renderResource}
        {...otherProps}
      />
    );
  }
}

export default rest.withResources(RestRefreshableScrollViewList);
