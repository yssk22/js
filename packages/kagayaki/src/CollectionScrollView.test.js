/* @flow */
import React from 'react';
import { Text } from 'react-native';
import CollectionScrollView from './CollectionScrollView';
import renderer from 'react-test-renderer';
import { rest } from '@yssk22/utakata';

type TestModel = {
  id: string,
  message: string
};

type TestResource = rest.Resource<TestModel>;

const renderResource = (v: TestResource) => {
  return <Text key={v.id}>{v.message}</Text>;
};

const settings = new rest.ResourceSettings('/foo/');

describe('CollectionScrollView', () => {
  const run = jest.fn();
  jest.spyOn(rest.Action, 'getCollection').mockImplementation(() => {
    return {
      run: run
    };
  });

  test('render while loading', () => {
    const props = rest
      .mockProps(settings)
      .status('reading')
      .mock();
    const component = renderer.create(
      <CollectionScrollView renderResource={renderResource} settings={settings} store={props} />
    );
    expect(rest.Action.getCollection).toHaveBeenCalled();
    expect(run).toHaveBeenCalled();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('render after loaded', () => {
    const props = rest
      .mockProps(settings)
      .data({
        id: '123',
        message: 'This is an item'
      })
      .status('none')
      .mock();
    const component = renderer.create(
      <CollectionScrollView renderResource={renderResource} settings={settings} store={props} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('render error after loaded', () => {
    const props = rest
      .mockProps(settings)
      .status('none')
      .error('this is error')
      .mock();
    const component = renderer.create(
      <CollectionScrollView renderResource={renderResource} settings={settings} store={props} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
