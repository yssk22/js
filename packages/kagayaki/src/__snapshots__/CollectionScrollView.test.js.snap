// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionScrollView render after loaded 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <RCTRefreshControl />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      This is an item
    </Text>
  </View>
</RCTScrollView>
`;

exports[`CollectionScrollView render error after loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    this is error
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        リトライ
      </Text>
    </View>
  </View>
</View>
`;

exports[`CollectionScrollView render while loading 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={true}
    />
  }
>
  <RCTRefreshControl />
  <View />
</RCTScrollView>
`;
